app/stores/ada/AdaTransactionBuilderStore.js

```diff
@@ -208,9 +209,22 @@ export default class AdaTransactionBuilderStore extends Store {
   }
 
   @action
-  _reset = async () => {
+  _preWaitReset = () => {
+    // resets that have to happen BEFORE waiting for createUnsignedTx to finish
     this.plannedTxInfo = [{ address: undefined, value: undefined }];
     this.shouldSendAll = false;
+  }
+  @action
+  _postWaitReset = () => {
+    // resets that have to happen AFTER waiting for createUnsignedTx to finish
+    this.createUnsignedTx.reset();
```

Comment:
Maybe it’s more clear in setup if we take out the createUnsignedTx from here

Reply:
So instead of just having pre and post, we would have pre, create and post

Reply:
where else would you put it? If we remove it from here, we then have to add it to `_reset` and `setup`

Reply:
Maybe we can tune up the names, it’s mostly that preWaitReset and postWaitReset are used inside Reset so the naming is a little bit tricky

Reply:
Maybe it could be something like: preWaitCreateUnsignedTx instead of preWaitReset